src/herder/HerderImpl.cpp

```diff
@@ -148,19 +148,37 @@ void
 HerderImpl::valueExternalized(uint64 slotIndex, StellarValue const& value)
 {
     const int DUMP_SCP_TIMEOUT_SECONDS = 20;
+    bool newSlot = slotIndex == getCurrentLedgerSeq();
 
-    // record metrics
-    getHerderSCPDriver().recordSCPExecutionMetrics(slotIndex);
+    if (newSlot)
+    {
+        getHerderSCPDriver().recordSCPExecutionMetrics(slotIndex);
```

Comment:
should this call to `HerderSCPDriver` be moved into `HerderSCPDriver::valueExternalized` ?